src/UI/Layout.hs

```diff
@@ -1,19 +1,27 @@
-{-# LANGUAGE FlexibleInstances, GADTs #-}
+{-# LANGUAGE FlexibleInstances, GADTs, ScopedTypeVariables, StandaloneDeriving, TypeOperators #-}
 module UI.Layout where
 
 import Control.Applicative
 import Control.Comonad.Cofree.Cofreer
 import Control.Monad.Free.Freer
+import Data.Fixed
 import Data.Functor.Algebraic
 import Data.Functor.Classes
 import Data.Functor.Foldable hiding (unfold)
+import Data.Functor.Listable
 import Data.Maybe (catMaybes, fromMaybe)
+import Data.Semigroup
+import Data.Typeable
 import UI.Geometry
 
+data Alignment = Leading | Trailing | Centre | Full
```

Comment:
The sizing behaviour prior to this PR can be recovered by using `Full` alignment (via the `alignFull` smart constructor); additionally, `Full` is the default.